app/models/manageiq/providers/ansible_tower/shared/automation_manager/configuration_script_source.rb

```diff
@@ -14,4 +14,18 @@ def provider_collection(manager)
   def provider_object(connection = nil)
     (connection || connection_source.connect).api.projects.find(manager_ref)
   end
+
+  REFRESH_ON_TOWER_SLEEP = 1.second
+  def refresh_in_provider
+    with_provider_object do |project|
+      return unless project.can_update?
+
+      project_update = project.update
+
+      while project_update.finished.blank?
```

Comment:
so it would be best to do it as an async job, with async waiting (probably processed by a generic worker), which would queue up a refresh once it's finished

unless I am mistaken, this does Ansible Tower full refresh, which we have seen to take 30+min with only hundreds of Vms. Maybe the numbers are better in the new Tower version